cpp/benchmarks/reduction/segment_reduce.cu

```diff
@@ -32,99 +32,98 @@
     dtype, distribution_id::UNIFORM, 0, 100);
   auto input = create_random_column(dtype, row_count{column_size}, profile);
 
-  auto offset_it =
-    detail::make_counting_transform_iterator(0, [column_size, segment_length] __device__(auto i) {
+  auto offset_it = cudf::detail::make_counting_transform_iterator(
+    0, [column_size, segment_length] __device__(auto i) {
       return column_size < i * segment_length ? column_size : i * segment_length;
     });
 
-  thrust::device_vector<size_type> d_offsets(offset_it, offset_it + num_segments + 1);
+  thrust::device_vector<cudf::size_type> d_offsets(offset_it, offset_it + num_segments + 1);
 
   return std::pair(std::move(input), d_offsets);
 }
 
-template <typename InputType, typename OutputType, aggregation::Kind kind>
-std::enable_if_t<!is_boolean_output_agg(kind) || std::is_same_v<OutputType, bool>, void>
-BM_Simple_Segmented_Reduction(nvbench::state& state,
-                              nvbench::type_list<InputType, OutputType, nvbench::enum_type<kind>>)
+template <typename DataType, cudf::aggregation::Kind kind>
+void BM_Simple_Segmented_Reduction(nvbench::state& state,
+                                   nvbench::type_list<DataType, nvbench::enum_type<kind>>)
 {
   // TODO: to be replaced by nvbench fixture once it's ready
   cudf::rmm_pool_raii rmm_pool;
 
-  auto const column_size{size_type(state.get_int64("column_size"))};
-  auto [input, offsets] = make_test_data<InputType>(state);
+  auto const column_size{cudf::size_type(state.get_int64("column_size"))};
+  auto const num_segments{cudf::size_type(state.get_int64("num_segments"))};
+
+  auto [input, offsets] = make_test_data<DataType>(state);
   auto agg              = make_simple_aggregation<kind>();
 
+  auto output_type = is_boolean_output_agg(kind) ? cudf::data_type{cudf::type_id::BOOL8}
+                                                 : cudf::data_type{cudf::type_to_id<DataType>()};
+
   state.add_element_count(column_size);
-  state.add_global_memory_reads<InputType>(column_size);
-  state.add_global_memory_writes<OutputType>(column_size);
+  state.add_global_memory_reads<DataType>(column_size);
+  if (is_boolean_output_agg(kind)) {
+    state.add_global_memory_writes<nvbench::int8_t>(num_segments);  // BOOL8
+  } else {
+    state.add_global_memory_writes<DataType>(num_segments);
+  }
+
+  auto const input_view  = input->view();
+  auto const offset_span = cudf::device_span<cudf::size_type>{offsets};
 
   state.exec(
-    nvbench::exec_tag::sync,
-    [input_view = input->view(), offset_span = device_span<size_type>{offsets}, &agg](
-      nvbench::launch& launch) {
-      segmented_reduce(
-        input_view, offset_span, *agg, data_type{type_to_id<OutputType>()}, null_policy::INCLUDE);
+    nvbench::exec_tag::sync, [input_view, output_type, offset_span, &agg](nvbench::launch& launch) {
+      segmented_reduce(input_view, offset_span, *agg, output_type, cudf::null_policy::INCLUDE);
```

Comment:
Similar to the implementation here: https://github.com/rapidsai/cudf/blob/63a47d91fc558a5b0b89f3ce9b0a61ffb3ed84f5/cpp/benchmarks/groupby/group_max.cpp#L55-L57
We need to pass the target stream to nvbench `state` thus performance metrics can be recorded properly. I'm a bit surprised that the execution didn't crash with the current implementation.